src/main/java/org/apereo/openlrs/controllers/caliper/CaliperController.java

```diff
@@ -86,28 +131,76 @@ public JsonNode getByIdHandler (
+            method = RequestMethod.GET,
+            value = {"", "/"},
+            params = {"!statementId",},
+            produces = "application/json;charset=utf-8"
+    )
+    @ResponseBody
+    public Page<JsonNode> getByFiltersHandler(
             @RequestParam(value = "actor", required = false) String actor,
             @RequestParam(value = "activity", required = false) String activity,
             @RequestParam(value = "since", required = false) String since,
             @RequestParam(value = "until", required = false) String until,
-            @RequestParam(value = "limit", required = false) String limit
-    ) throws InvalidRequestException {
-        try {
-            Map<String, String> filterMap = null;
-            filterMap = StatementUtils.createStatementFilterMap(actor, activity, since, until, limit);
-            return caliperService.getJsonNodes(filterMap);
-        } catch (Exception e) {
-            logger.error(e.getMessage(), e);
-            throw new InvalidRequestException(e.getMessage(), e);
+            @RequestParam(value = "page", required = false) String pageParam,
+            @RequestParam(value = "limit", required = false) String limitParam,
+            @RequestParam(value = "number", required = false) String numberParam,
+            @RequestParam(value = "size", required = false) String sizeParam
+    ) throws NumberFormatException {
+        Map<String, String> filterMap =
+                StatementUtils.createStatementFilterMap(actor, activity, since, until, limitParam);
+        Integer page = this.getPageDefault();
+        String pageParamName = "page";
+        Integer limit = this.getLimitDefault();
+        String limitParamName = "limit";
+        Integer limitMaximum = this.getLimitMaximum();
+
+        if ((pageParam == null) && (numberParam != null)) {
+            pageParam = numberParam;
+            pageParamName = "number";
         }
+
+        if (pageParam != null) {
+            try {
+                page = Integer.parseInt(pageParam);
+
+                if (page < 0) {
+                    throw new NumberFormatException("For input string: \"" + pageParam + "\"");
+                }
+            } catch (NumberFormatException e) {
+                throw new NumberFormatException(pageParamName + " must be a positive integer (" + e.getMessage() + ")");
+            }
+        }
+
+        if ((limitParam == null) && (sizeParam != null)) {
+            limitParam = sizeParam;
+            limitParamName = "size";
+        }
+
+        if (limitParam != null) {
+            try {
+                limit = Integer.parseInt(limitParam);
+
+                if ((limit < 1) || (limit > limitMaximum)) {
+                    throw new NumberFormatException("For input string: \"" + limitParam + "\"");
+                }
+            } catch (NumberFormatException e) {
+                throw new NumberFormatException(limitParamName + " must be an integer between 1 and " +
```

Comment:
Again - this is an expensive way to handle an error - just throw the exception once